braindecode/datautil/windowers.py

```diff
@@ -86,23 +104,28 @@ def create_windows_from_events(
             )
 
         events, events_id = mne.events_from_annotations(ds.raw, mapping)
-        onsets = events[:, 0]
+        onsets = events[:, 0] - ds.raw.first_samp
         filtered_durations = np.array(
-            [a['duration'] for a in ds.raw.annotations if a['description'] in events_id]
+            [a['duration'] for a in ds.raw.annotations
+             if a['description'] in events_id]
         )
         stops = onsets + (filtered_durations * ds.raw.info['sfreq']).astype(int)
+        # XXX This could probably be simplified by using chunk_duration in
+        #     `events_from_annotations`
 
-        if stops[-1] + trial_stop_offset_samples > len(ds):
+        if (stops[-1] + trial_stop_offset_samples >
+                ds.raw.n_times + ds.raw.first_samp):
```

Comment:
This seems potentially quite errorprone code, do we have a test for this?

Reply:
Yes, `test_windowers.py::test_windows_cropped` used to test this indirectly - I'll add something more specific to this catch.

Reply:
Actually I'm not 100% sure the behaviour of `mne.Epochs` is correct when the raw file has been cropped. Maybe there's something I'm missing that would make the handling of `raw.first_samp` simpler... I opened an issue here https://github.com/mne-tools/mne-python/issues/8201 just in case.

Reply:
So now this will be correct, but need newest mne from master? just for clarification